03week/towersOfHanoi.js

```diff
@@ -13,33 +13,90 @@ let stacks = {
   c: []
 };
 
-function printStacks() {
+let moves = 0;//to track moves so we only have to check for win after 14
+
+const printStacks=()=> {
+  console.log(`Moves: ${moves}`.cyan.underline);//log moves for every turn in cyan and underlined
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
+const isInputValid=(startStack, endStack)=> {//set tests that user input is valid
+  const validInputs = /[abc]/;//valid inputs
+  if (startStack.length === 1 && endStack.length === 1 &&
```

Comment:
can be simplified to `return startStack.length === 1 && endStack.length === 1 && startStack.search(validInputs) === 0 && endStack.search(validInputs) === 0` and the function that calls this function would be responsible for the announcement to the user. This functions ONLY purpose should be to determine if the input is valid